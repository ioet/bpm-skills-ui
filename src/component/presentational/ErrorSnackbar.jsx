import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Snackbar from '@material-ui/core/Snackbar/Snackbar';
import PropTypes from 'prop-types';
import { ErrorSnackbarConst } from '../../constants';
import { SnackbarStyles } from '../../styles';

const ErrorSnackbar = (props) => {
  const {
    classes, open, handleClose, message,
  } = props;

  return (
    <Snackbar
      className={classes.snackbar}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      open={open}
      autoHideDuration={ErrorSnackbarConst.AUTO_HIDE_DURATION}
      onClose={handleClose}
      ContentProps={{
        'aria-describedby': 'message-id',
      }}
      message={(
        <span id="message-id">
          {message}
        </span>
      )}
    />
  );
};

ErrorSnackbar.defaultProps = {
  message: '',
};

ErrorSnackbar.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  message: PropTypes.string,
};

export default withStyles(SnackbarStyles)(ErrorSnackbar);
