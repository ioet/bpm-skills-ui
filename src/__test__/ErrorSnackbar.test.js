import React from 'react';
import ErrorSnackbar from '../component/presentational/ErrorSnackbar';

describe('<ErrorSnackbar />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<ErrorSnackbar />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
