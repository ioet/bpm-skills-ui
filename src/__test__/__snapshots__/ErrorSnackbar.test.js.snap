// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorSnackbar /> Component Renders correctly 1`] = `
<ErrorSnackbar
  classes={
    Object {
      "snackbar": "ErrorSnackbar-snackbar-1",
    }
  }
  message=""
/>
`;
